src/installer/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -293,7 +293,8 @@ public string GenerateBundle(IReadOnlyList<FileSpec> fileSpecs)
             // We will memory map a larger file than needed, but we'll take that trade-off.
             foreach (var (spec, type) in relativePathToSpec)
             {
-                bundledFilesSize += new FileInfo(spec.SourcePath).Length;
+                using (var stream = File.OpenRead(spec.SourcePath))
+                    bundledFilesSize += stream.Length;
```

Comment:
Without commenting on the underlying issue I think it would be better to avoid opening the file here.  Something like the following seems more targeted.
```suggestion
                var info = new FileInfo(spec.SourcePath);
                bundledFilesSize += (info.LinkTarget is null ? info : new FileInfo (info.LinkTarget)).Length;
```

Reply:
I believe that'll only follow one link?

Reply:
probably correct, try File.ResolveLinkTarget

Reply:
Just FYI there are currently two places in runtime that do this:

```
        // It means we are dealing with a symlink and need to get the information
        // from the target file instead.
        if (fileInfo.Exists && !string.IsNullOrEmpty(fileInfo.LinkTarget))
        {
            fileInfo = (FileInfo?)fileInfo.ResolveLinkTarget(returnFinalTarget: true) ?? fileInfo;
        }
```

I think I'd prefer to do `((FileInfo)info.ResolveLinkTarget(true) ?? info).Size`.

This seems like something that should be a helper, but maybe just locally?